### PR TITLE
dev-python/ptpython: bump to version 2.0.6.

### PR DESCRIPTION
**Pull request by dargor**
Also bump `dev-python/prompt_toolkit`, needed.

Version 3.x of `dev-python/prompt_toolkit` will require at least python 3.6, so stick to version 2.x for now. It's excluded by upstream requirements anyway.

### CODE REVIEW

File: dev-python/prompt_toolkit/prompt_toolkit-2.0.10.ebuild [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+RDEPEND="
+	>=dev-python/six-1.9.0[${PYTHON_USEDEP}]
+	dev-python/wcwidth[${PYTHON_USEDEP}]"
+
+DEPEND="
+	${RDEPEND}
+	test? ( dev-python/pytest[${PYTHON_USEDEP}] )"
+
+python_test() {
+	py.test || die
```
Comment (wimmuskee) [inline dev-python/prompt_toolkit/prompt_toolkit-2.0.10.ebuild]: You can replace the test USE/RESTRICT, pytest dep and the python_test function with the _distutils_enable_tests pytest_ call from the distutils-r1 eclass.

File: dev-python/ptpython/ptpython-2.0.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+RDEPEND="
+	dev-python/docopt[${PYTHON_USEDEP}]
+	>=dev-python/jedi-0.9.0[${PYTHON_USEDEP}]
+	>=dev-python/prompt_toolkit-2.0.8[${PYTHON_USEDEP}]
+	<dev-python/prompt_toolkit-2.1.0[${PYTHON_USEDEP}]
+	dev-python/pygments[${PYTHON_USEDEP}]
+"
+
+# Not included
+RESTRICT=test
```
Comment (flappyports) [inline dev-python/ptpython/ptpython-2.0.6.ebuild]: Why are you restricting tests here, but leaving the function for running tests below?

File: dev-python/ptpython/ptpython-2.0.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+python_test() {
+	"${PYTHON}" tests/run_tests.py || die
```
Comment (flappyports) [inline dev-python/ptpython/ptpython-2.0.6.ebuild]: This test file doesn't exist and you are restricting tests above (as the src does not provide any tests).

File: dev-python/ptpython/ptpython-2.0.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 python3_{5,6,7} )
```
Comment (flappyports) [inline dev-python/ptpython/ptpython-2.0.6.ebuild]: Please drop py2.7 support from all. Gentoo is beginning the process of removing py2 support

File: dev-python/ptpython/ptpython-2.0.6.ebuild (hunk 2 above)
Comment (flappyports) [inline dev-python/ptpython/ptpython-2.0.6.ebuild]: Also, please move the RESTRICT to below IUSE

File: dev-python/ptpython/ptpython-2.0.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="
+	dev-python/docopt[${PYTHON_USEDEP}]
+	>=dev-python/jedi-0.9.0[${PYTHON_USEDEP}]
+	>=dev-python/prompt_toolkit-2.0.8[${PYTHON_USEDEP}]
+	<dev-python/prompt_toolkit-2.1.0[${PYTHON_USEDEP}]
```
Comment (flappyports) [inline dev-python/ptpython/ptpython-2.0.6.ebuild]: Why the version constraints here? I could understanding >= but I am not tracking a reason for <2.1.0

File: dev-python/prompt_toolkit/prompt_toolkit-2.0.10.ebuild [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 python3_{5,6,7} )
```
Comment (flappyports) [inline dev-python/prompt_toolkit/prompt_toolkit-2.0.10.ebuild]: Please drop py2.7 support

File: dev-python/ptpython/ptpython-2.0.6.ebuild (hunk 5 above)
Comment (dargor) [reply dev-python/ptpython/ptpython-2.0.6.ebuild]: Simply following upstream requirements in `setup.py`, the versions of prompt_toolkit can break a lot of things and you want to keep these in sync.

**Pull request closed without merging**